sys/aarch64/rootdev.c

```diff
@@ -302,21 +304,27 @@ static bus_methods_t rootdev_bus_if = {
 static driver_t rootdev_driver = {
   .size = sizeof(rootdev_t),
   .desc = "RPI3 platform root bus driver",
+  .probe = rootdev_probe,
   .attach = rootdev_attach,
-  .interfaces = {
-    [DIF_BUS] = &rootdev_bus_if,
-  },
+  .interfaces =
+    {
+      [DIF_BUS] = &rootdev_bus_if,
+    },
+  .pass = FIRST_PASS,
 };
 
-static device_t rootdev = (device_t){
-  .children = TAILQ_HEAD_INITIALIZER(rootdev.children),
-  .driver = (driver_t *)&rootdev_driver,
-  .state = &(rootdev_t){},
-  .devclass = &DEVCLASS(root),
-};
-
-DEVCLASS_ENTRY(root, rootdev);
+DEVCLASS_CREATE(root);
 
-void init_devices(void) {
-  device_attach(&rootdev);
+void init_devices(pass_num_t pass) {
```

Comment:
Please note that first pass is performed when `rootdev` is `NULL`. Other passes when `rootdev` is already present.

Reply:
Done.